### PR TITLE
Exploration: publish integration test suite as separate package (version 4)

### PR DESCRIPTION
**Pull request by trevor-scheer**
Related exploration: exporting the test suite from `@apollo/server` itself https://github.com/apollographql/apollo-server/pull/6617

The exported test suite resulted in a dependency burden for both integration authors _and_ consumers, which is really a large imposition to make on all projects who want to use Apollo Server. With the test suite as a separate package, it's something that authors can opt in to use if they want to ensure parity with the core project.

Fortunately for us, Jest allows us to just export a test suite as a package, import it elsewhere, and run the tests. To confirm this all works as intended, I:
* Installed the server and test suite codesandbox builds from this PR into a separate project
* Configured jest, created a test file, imported `defineIntegrationTestSuite`
* Duplicated how `defineIntegrationTestSuite` is being called for express within this repo
* Ran tests to confirm all passing (140), confirmed this number in the main repo
* "Broke" configuration of the app to incite test failures to ensure failures happen when expected

TODO
- [ ] Should I export the types for the signature? They're baked in to the function, so aren't really needed. Possibly helpful though.
- [ ] Is there a test I should be running _on_ the test suite export itself? I suspect not necessary.

### CODE REVIEW

**Comment by trevor-scheer** [comment]
@glasser merging, please see TODOs with questions we can address in a follow-up PR.

File: packages/server/.npmignore [hunk 1]
```diff
@@ -1,6 +1,7 @@
 *
 # Include the package.json files that give us deep imports.
 !**/package.json
+# TODO(AS4): glasser isn't this covered by the * above?
```
Comment (glasser) [inline packages/server/.npmignore:4]: hmm, the previous line will pull `node_modules/foo/package.json` back in, right?

File: packages/integration-testsuite/tsconfig.json [hunk 2]
```diff
@@ -0,0 +1,12 @@
+{
```
Comment (glasser) [inline packages/integration-testsuite/tsconfig.json:1]: Maybe worth a comment in this file or elsewhere noting that we don't dual-build the integration test suite. (Why? I guess the advantage of ESM is that bundlers can be smarter, and nobody is bundling the test suite into their apps. Also maybe some of our test deps don't support ESM? Worth noting though.0

**Review by glasser — COMMENTED** [review]
> Should I export the types for the signature? They're baked in to the function, so aren't really needed. Possibly helpful though.

Either way seems fine. Why not, I guess.

> Is there a test I should be running on the test suite export itself? I suspect not necessary.

I suppose you could have smoke-test run it? Might be overkill though.

**Pull request merged**